frontend/src/metabase/static-viz/components/XYChart/types.ts

```diff
@@ -15,13 +16,24 @@ export type YAxisPosition = "left" | "right";
 
 export type VisualizationType = "line" | "area" | "bar" | "waterfall";
 
-export type Series = {
+interface BaseSeries {
   name: string;
-  color: string;
   data: SeriesData;
   type: VisualizationType;
   yAxisPosition: YAxisPosition;
-};
+}
+
+export interface SeriesWithoutBreakoutValues extends BaseSeries {
+  seriesKey: string;
+}
+
+export interface SeriesWithBreakoutValues extends BaseSeries {
```

Comment:
Technically, we breakdown metrics so that series can be a metric for all values or for a single value. Maybe we can adjust the naming of these

Reply:
This might be irrelevant soon as I'm making changes to the API, but this initially was meant for a series with 2 dimensions—side question can there be more than 2 dimensions e.g. 3 or 4 as I don't think it's possible in the UI?